presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -64,6 +64,7 @@
     private final Optional<Symbol> rightHashSymbol;
     private final Optional<DistributionType> distributionType;
     private Optional<Boolean> spatialJoin = Optional.empty();
```

Comment:
what about explicit `sourceId`?

Reply:
Done !!.. Adding DynamicFilterSource instead of 2 fields sounded better.